pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -57,6 +58,18 @@ public PgDatabaseMetaData(PgConnection conn) {
   private int nameDataLength; // length for name datatype
   private int indexMaxKeys; // maximum number of keys in an index.
 
+  private byte[] getCatalogName(@Nullable String catalog) throws SQLException {
+    if (catalog == null) {
+      try {
+        return connection.encodeString(connection.getCatalog());
+      } catch (SQLException e) {
+        // If we can't get the catalog name, we'll just use the empty string
+        return new byte[0];
+      }
+    }
+    return catalog.getBytes(Charset.defaultCharset());
+  }
+
```

Comment:
From what I can tell we check to see if the catalog is null before calling this function. 
Moot point: If we get a SQLException we should just throw it.

Reply:
There are many checks like the following in the code:

```java
if (catalog != null) {
  sql += " AND current_database() = " + escapeQuotes(catalog);
}
```
They will not exclude `null`. Likewise, the additional checks in #3588 don't exclude `catalog` to be `null` afaik. 

Removing the `try`/`catch` makes sense as errors can bubble up to the application.

Reply:
In retrospect I should have removed all of those in https://github.com/pgjdbc/pgjdbc/pull/3588. By the time we get to the query we know that if the catalog is not null then it is the same as `current_database()` If you want to make those changes I'd appreciate it. Otherwise I'll make them which might make your's harder to rebase.

Reply:
I'll take a look at it!